lib/azure/armrest/model/base_model.rb

```diff
@@ -62,7 +74,20 @@ def initialize(json)
           @json = json
         end
 
-        @ostruct = OpenStruct.new(hash)
+        ostruct = AzureOpenStruct.new(hash)
+        keys = ostruct.to_h.keys
+
+        # Override the Delegate's normal behavior so that it any name collisions
+        # become methods for our model rather than skipped. An _alias is created
```

Comment:
That approach basically means we should always use hash style syntax within the armrest gem then, because we could never know in advance if an external library that requires it has defined custom Object or BasicObject methods.